core/src/test/java/apoc/periodic/PeriodicTest.java

```diff
@@ -51,21 +53,64 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 import static org.neo4j.driver.internal.util.Iterables.count;
+import static org.neo4j.test.assertion.Assert.assertEventually;
 
 public class PeriodicTest {
 
     public static final long RUNDOWN_COUNT = 1000;
     public static final int BATCH_SIZE = 399;
 
+    public static AssertableLogProvider logProvider = new AssertableLogProvider();
+
     @Rule
-    public DbmsRule db = new ImpermanentDbmsRule();
+    public DbmsRule db = new ImpermanentDbmsRule(logProvider);
 
     @Before
     public void initDb() {
         TestUtil.registerProcedure(db, Periodic.class, Schemas.class, Cypher.class, Utils.class);
         db.executeTransactionally("call apoc.periodic.list() yield name call apoc.periodic.cancel(name) yield name as name2 return count(*)");
     }
 
+    @Test
+    public void testRepeatWithVoidProcedure() {
+        String query = "CALL apoc.periodic.repeat('repeat-1', 'CALL db.prepareForReplanning()', 1)";
+        testLogIncrease(query);
+    }
+
+    @Test
+    public void testRepeatWithVoidProcedureAndReturn() {
+        String query = "CALL apoc.periodic.repeat('repeat-2', 'CALL db.prepareForReplanning() RETURN 1', 1)";
+        testLogIncrease(query);
+    }
+
+    @Test
+    public void testSubmitWithVoidProcedure() {
+        String query = "CALL apoc.periodic.submit('submit-1', 'CALL db.prepareForReplanning() RETURN 1')";
+        testLogIncrease(query);
+    }
+
+    @Test
+    public void testSubmitWithVoidProcedureAndReturn() {
+        String query = "CALL apoc.periodic.submit('submit-1', 'CALL db.prepareForReplanning()')";
+        testLogIncrease(query);
+    }
+
+    private void testLogIncrease(String query) {
+        // clear logs to prevent potential false positives
+        logProvider.clear();
+
+        // execute a periodic procedure with `CALL db.prepareForReplanning` as an inner procedure
+        db.executeTransactionally(query);
+
+        // print logs in logProvider
+        // and check that the previous `prepareForReplanning` logs a "Manual trigger for sampling all indexes"
+        assertEventually(() -> {
+            logProvider.print(System.out);
+            return logProvider.serialize().contains("Manual trigger for sampling all indexes");
```

Comment:
We are relying on something we write to the logs here in a procedure we don't control. We just need a void procedure with some effect right? 

I feel like it would be a better approach if we do a `dbms.setConfigValue` for example, fetching the value of that config afterwards.